doc/development_process/documentation.rst

```diff
@@ -0,0 +1,87 @@
+
+Code Documentation
+###################
+
+API Documentation
+******************
+
+Well documented APIs enhance the experience for developers and are an essential
+requirement for defining an API's success. Doxygen is a general purpose
+documentation tool that the Zephyr project uses for documenting APIs. It
+generates an on-line documentation browser (in HTML) and provides
+input for other tools used to generate a reference manual from
+documented source files.
```

Comment:
Maybe add:

`In particular, doxygen's XML output is used as an input when producing the Zephyr project's online documentation.`